pkg/genlib/generator_interface.go

```diff
@@ -617,7 +658,46 @@ func fuzzyFloat(previous, fuzziness, min, max float64) float64 {
 	return lowerBound + customRand.Float64()*(higherBound-lowerBound)
 }
 
+func fuzzyFloatCounter(previous, fuzziness float64) float64 {
+	lowerBound := previous
+	higherBound := previous * (1 + fuzziness)
+	return lowerBound + customRand.Float64()*(higherBound-lowerBound)
+}
+
 func bindDouble(fieldCfg ConfigField, field Field, fieldMap map[string]any) error {
+	if err := fieldCfg.ValidCounter(); err != nil {
+		return err
+	}
+
+	if fieldCfg.Counter {
+		var emitFNotReturn emitFNotReturn
+		emitFNotReturn = func(state *genState, buf *bytes.Buffer) error {
+			var previous float64
+			var dummyFloat float64
+			var dummyFunc func() float64
+
+			if previousDummyFloat, ok := state.prevCache[field.Name].(float64); ok {
+				previous = previousDummyFloat
+			}
+
+			if fieldCfg.Fuzziness <= 0 {
+				dummyFunc = makeFloatCounterFunc(previous, field)
+
+				dummyFloat = dummyFunc()
+			} else {
+				dummyFloat = fuzzyFloatCounter(previous, fieldCfg.Fuzziness)
+			}
+
+			state.prevCache[field.Name] = dummyFloat
+			_, err := fmt.Fprintf(buf, "%f", dummyFloat)
```

Comment:
Is this to catch potential conversion errors?

Reply:
https://pkg.go.dev/fmt#Fprintf

> err return values from Fprintf are those returned by the underlying io.Writer.